codec-http2/src/main/java/io/netty/handler/codec/http2/DefaultHttp2Connection.java

```diff
@@ -833,7 +834,8 @@ public Http2Stream closeRemoteSide() {
      */
     private final class DefaultEndpoint<F extends Http2FlowController> implements Endpoint<F> {
         private final boolean server;
-        private int nextStreamId;
+        private int nextExpectedStreamId;
```

Comment:
> nextExpectedStreamId is the minimum stream ID allowed for the next stream creation. This is generally equivalent to lastStreamCreated + 1 (assuming that lastStreamCreated has been set).

Yes (`lastStreamCreated + 2` though). I think we can remove the `lastStreamCreated` variable and just compute it as needed.

> Also I wonder if renaming could help make it more obvious:

Sure I will rename and add comments.